application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -222,6 +231,25 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                     schemaUpdateFile = Paths.get(installScripts.getDataDir(), "upgrade", "3.1.0", SCHEMA_UPDATE_SQL);
                     loadSql(schemaUpdateFile, conn);
                     log.info("Schema updated.");
+                    try {
+                        conn.createStatement().execute(
+                                "ALTER TABLE device ADD entity_profile_id varchar(31);"); //NOSONAR, ignoring because method used to execute thingsboard database upgrade script
+                    } catch (Exception e) {
+                        log.warn(ExceptionUtils.getRootCauseMessage(e));
+                    }
+                    transactionTemplate.execute(transactionStatus -> {
+                        deviceRepository.findAllTenantIdAndTypeWhereEmptyEntityProfileId().forEach(objects -> {
+                            String tenantIdStr = (String) objects[0];
+                            TenantId tenantId = new TenantId(UUIDConverter.fromString(tenantIdStr));
+                            String type = (String) objects[1];
+
+                            EntityProfile profile = tbEntityProfileService.createDefault(tenantId, type, DeviceProfile.class);
+                            String entityProfileStr = UUIDConverter.fromTimeUUID(profile.getId().getId());
+                            deviceRepository.updateEntityProfileId(entityProfileStr, tenantIdStr, type);
```

Comment:
Whey do we use entityProfileStr, tenantIdStr instead of TenantId and EntityProfileId?

Reply:
because I use directly deviceRepository and DeviceEntity has tenantId and entityProfileId as string
no needed add this methods for service layer